lib/Runtime/Base/CompactCounters.h

```diff
@@ -11,32 +11,40 @@ namespace Js
 {
     template<class T, typename CountT = T::CounterFields>
     struct CompactCounters
-    {
+    {    
+        template<typename CountT>
+        struct Fields {
+            union {
+                uint8 u8Fields[CountT::Max];
+                int8 i8Fields[CountT::Max];
+                uint16 u16Fields[CountT::Max];
+                int16 i16Fields[CountT::Max];
+                uint32 u32Fields[CountT::Max];
+                int32 i32Fields[CountT::Max];
+            };
+            Fields() {}
+        };
+
+        typedef Fields<CountT> FieldsT;
+
         uint8 fieldSize;
 #if DBG
+    
         mutable bool bgThreadCallStarted;
         bool isCleaningUp;
-#endif
-        union {
-            uint8 u8Fields[CountT::Max];
-            int8 i8Fields[CountT::Max];
-            WriteBarrierPtr<uint16> u16Fields;
-            WriteBarrierPtr<uint32> u32Fields;
-            WriteBarrierPtr<int16> i16Fields;
-            WriteBarrierPtr<int32> i32Fields;
-        };
-
-        CompactCounters()
-            :fieldSize(1)
+#endif    
+        WriteBarrierPtr<Fields<CountT>> fields;
```

Comment:
good catch. don't need template parameter for Fields at all. I moved it out and moved it in so forget to remove the template parameter